### ISSUE TITLE
Test decimal::tests::to_double panicks on i386

### ISSUE THREAD

**Issue by jonassmedegaard** [oxigraph/oxigraph#930]
With Oxigraph v0.4.0-alpha.7, patched to include git commits 9c0b303 and ed2b4e5, tests fail on Debian i386 arch like this:
```
1552s failures:
1552s 
1552s ---- decimal::tests::to_double stdout ----
1552s thread 'decimal::tests::to_double' panicked at src/decimal.rs:1063:9:
1552s assertion `left == right` failed
1552s   left: Double { value: 0.1 }
1552s  right: Double { value: 0.1 }
1552s stack backtrace:
1552s    0: rust_begin_unwind
1552s    1: core::panicking::panic_fmt
1552s    2: core::panicking::assert_failed_inner
1552s    3: core::panicking::assert_failed
1552s    4: oxsdatatypes::decimal::tests::to_double
1552s    5: core::ops::function::FnOnce::call_once
```
Full build log is [here](https://ci.debian.net/data/autopkgtest/testing/i386/o/oxigraph/49206522/log.gz)

**Comment by jonassmedegaard** [comment]
Possibly relevant: It was an optimized build (i.e. with options similar to --release passed also for building tests).

**Comment by Tpt** [comment]
Released as part of 0.4.0-alpha.8
